### PR TITLE
Let user decide whether animation auto stop/start when app resign/resume active

### PR DESCRIPTION
**Pull request by eJamesLin**
Add bool value to let sdk user decide whether animation should be
- automatically stopped before resign active
- automatically started after resume active

### CODE REVIEW

File: APNGKit/APNGImageView.swift [hunk 1]
```diff
@@ -150,7 +158,7 @@ open class APNGImageView: APNGView {
         
         addObservers()
```
Comment (Byounggon) [inline APNGKit/APNGImageView.swift:159]: It is unnecessary code when both autoStopAnimationBeforeResignActive and autoStartAnimationAfterResumeActive are false.
How about it?
`private var isObserverNeeded: Bool {
        autoStopAnimationBeforeResignActive || autoStartAnimationAfterResumeActive
}`
`if isObserverNeeded {
    addObservers()
}`

Comment (eJamesLin) [reply APNGKit/APNGImageView.swift:159]: @Byounggon `addObservers` is at initializer, and also the required `init(coder:)` and `init(frame:)` is difficult to have default value of `autoStopAnimationBeforeResignActive` `autoStartAnimationAfterResumeActive` parameters before initialization

**Review by Byounggon — APPROVED** [review]
LGTM

**Review by Byounggon — APPROVED** [review]

**Comment by onevcat** [comment]
@eJamesLin 
Maybe let's add back the `autoStartAnimationAfterResumeActive` so we can keep align the behavior for both sides.

**Comment by onevcat** [comment]
After some more consideration, `autoStopAnimationBeforeResignActive` flag would make it quite complicated when determine the `wasAnimating`. So let keep it as it is for now!

**Pull request merged**